src/jemalloc.c

```diff
@@ -3713,10 +3714,22 @@ je_malloc_stats_print(void (*write_cb)(void *, const char *), void *cbopaque,
 
 	tsdn = tsdn_fetch();
 	check_entry_exit_locking(tsdn);
-	stats_print(write_cb, cbopaque, opts);
+	if (write_cb != NULL || je_malloc_message != NULL) {
+		stats_print(write_cb, cbopaque, opts);
+	} else {
```

Comment:
With this change, how do we test the buffered path? IIRC, the tests all customize the two things above?